### PR TITLE
feat(docs): add markdownlint

### PR DESCRIPTION
**Pull request by 0xSooki**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Description

Implemented markdownlint to lint the docs folder, added the package itself & 2 scripts to the package.json to be able to lint & fix with markdownlint

## Added tests?

- [ ] Yes
- [x] No, because they aren't needed

## Added to documentation?

- [ ] README
- [ ] Documentation
- [x] No Documentation Needed

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,9 @@
     "test:build": "run-s build",
     "test:unit": "jest",
     "test:lint": "eslint src/**/*.ts src/**/*.tsx --fix",
-    "test:unit:watch": "jest --watch"
+    "test:unit:watch": "jest --watch",
+    "lint:markdown": "markdownlint './docs/**/*.md' --ignore node_modules -c .markdownlint.json",
```
Comment (darkmatter18) [inline package.json:29]: Add it in the `test:lint` with eslint

**Comment by darkmatter18** [comment]
Your commit messages are not conventionally supported.

File: package.json (hunk 1 above)
Comment (0xSooki) [reply package.json:29]: Gotcha, working on it

**Comment by 0xSooki** [comment]
Updated commit messages to be conventionally supported

**Pull request merged**